lib/Frontend/Frontend.cpp

```diff
@@ -138,70 +158,122 @@ bool CompilerInstance::setup(const CompilerInvocation &Invok) {
 
     Context->addModuleLoader(std::move(clangImporter), /*isClang*/ true);
   }
+  return false;
+}
 
-  assert(Lexer::isIdentifier(Invocation.getModuleName()));
-
-  Optional<unsigned> CodeCompletionBufferID;
-  auto CodeCompletePoint = Invocation.getCodeCompletionPoint();
-  if (CodeCompletePoint.first) {
-    auto MemBuf = CodeCompletePoint.first;
+Optional<unsigned> CompilerInstance::setupCodeCompletionBuffer() {
+  Optional<unsigned> codeCompletionBufferID;
+  auto codeCompletePoint = Invocation.getCodeCompletionPoint();
+  if (codeCompletePoint.first) {
+    auto memBuf = codeCompletePoint.first;
     // CompilerInvocation doesn't own the buffers, copy to a new buffer.
-    CodeCompletionBufferID = SourceMgr.addMemBufferCopy(MemBuf);
-    InputSourceCodeBufferIDs.push_back(*CodeCompletionBufferID);
-    SourceMgr.setCodeCompletionPoint(*CodeCompletionBufferID,
-                                     CodeCompletePoint.second);
+    codeCompletionBufferID = SourceMgr.addMemBufferCopy(memBuf);
+    InputSourceCodeBufferIDs.push_back(*codeCompletionBufferID);
+    SourceMgr.setCodeCompletionPoint(*codeCompletionBufferID,
+                                     codeCompletePoint.second);
   }
+  return codeCompletionBufferID;
+}
 
-  bool MainMode = (Invocation.getInputKind() == InputFileKind::IFK_Swift);
-  bool SILMode = (Invocation.getInputKind() == InputFileKind::IFK_SIL);
+bool CompilerInstance::setupInputs(Optional<unsigned> codeCompletionBufferID) {
+  // Add the memory buffers first, these will be associated with a filename
+  // and they can replace the contents of an input filename.
```

Comment:
This comment is no longer correct/relevant.

Reply:
Deleted.